client-v2/src/shared/components/input/CreateInput.tsx

```diff
@@ -44,20 +52,59 @@ export default (
     React.HTMLAttributes<HTMLInputElement> & StyledProps<any>
   >,
   type?: string
-) => ({ label, input, originalValue, ...rest }: Props) => (
-  <InputContainer>
-    {label && (
-      <InputLabel htmlFor={label}>
-        {label}
-        {originalValue && input.value !== originalValue && (
-          <RewindButton onClick={() => input.onChange(originalValue)}>
-            <RewindIcon />
-          </RewindButton>
-        )}
-      </InputLabel>
-    )}
-    <InputComponentContainer>
-      <Component id={label} {...input} {...rest} type={type} />
-    </InputComponentContainer>
-  </InputContainer>
-);
+) => {
+  return class extends React.Component<Props, State> {
+    private inputElement: React.RefObject<HTMLInputElement>;
+    constructor(props: Props) {
+      super(props);
+      this.inputElement = React.createRef();
+      this.state = { inputHeight: 36 };
+    }
+
+    public componentDidMount() {
+      this.updateHeight();
+    }
+
+    public componentDidUpdate(prevProps: Props) {
+      if (prevProps !== this.props) {
```

Comment:
As an optimisation, could we go one better here, and only update the height if the prop that is likely to change it (the input value) has changed?